### ISSUE TITLE
DataFrame.koalas.apply_batch seems to repartition

### ISSUE THREAD

**Issue by jimfulton** [databricks/koalas#1779]
I'm dealing with a case where I need to keep partitions small. If I don't, I get lots of errors relating to memory usage (exceeding memory limits, PyArrow length errors, NullPointer exceptions).

In my applicatiion, I'm repartitiing data to 800 partitions, but the output of `apply_batch` has 200 partitions.  Sometimes, repartiting that again works, but sometimes the repartioning fails (e.g. `TaskKilled (Stage cancelled)
20/09/16 13:55:12 WARN TaskSetManager: Lost task 173.3 in stage 7.0 (TID 1913, cilhadps0144.sys.cigna.com, executor 11): java.lang.NullPointerException`)

I'm using Spark 2 (2.4.4).

I'm surprised the batches used aren't partitions. 

Is there any I can control the number of partitions output by apply_batch?

**Comment by ueshin** [comment]
You can control the partition number for shuffle in general by a Spark config `spark.sql.shuffle.partitions`. https://spark.apache.org/docs/latest/sql-performance-tuning.html, but I'd recommend to use Spark 3.0 where no shuffle is needed during `apply_batch`.

For `NullPointerException`, do you have any simple example to reproduce it?

**Comment by jimfulton** [comment]
Thanks! I was just looking for a setting like that.

I'm looking forward to Spark 3. :) It doesn't seem to be supported by AWS Glue or EMR.  I'm hopeful I can make it work in our on-prem cluster.

WRT NullPointer exception, no simple example, sorry.  IN the dataset I'm messing with, I got no error with 3 million records, then got the error with 4 million, then made it go away by repartitioning the output, and then it came back when I tried 17 million records.
